File: apps/android/feature/workouts/domain/src/main/java/com/scribblefit/feature/workouts/domain/usecase/CalculateWorkoutVolumeUseCase.kt
package com.scribblefit.feature.workouts.domain.usecase

import com.scribblefit.core.model.Workout

class CalculateWorkoutVolumeUseCase {
    operator fun invoke(workout: Workout): Double {
        return workout.exercises.sumOf { exercise ->
            exercise.sets.sumOf { set ->
                (set.weight?.toDouble() ?: 0.0) * set.reps
            }
        }
    }
}

File: apps/ios/ScribbleFit/LocalPackages/Features/FeatureWorkouts/Sources/FeatureWorkouts/Domain/UseCases/CalculateWorkoutVolumeUseCase.swift
import Foundation
import CoreModel

public struct CalculateWorkoutVolumeUseCase: Sendable {
    public init() {}
    
    public func execute(workout: Workout) -> Double {
        workout.exercises.reduce(0.0) { workoutTotal, exercise in
            workoutTotal + exercise.sets.reduce(0.0) { exerciseTotal, set in
                exerciseTotal + Double((set.weight ?? 0) * Float(set.reps))
            }
        }
    }
}
